src/mono/mono/component/debugger-agent.c

```diff
@@ -9338,8 +9338,8 @@ frame_commands (int command, guint8 *p, guint8 *end, Buffer *buf)
 			} else {
 				MonoDebugLocalsInfo *locals;
 
-				locals = mono_debug_lookup_locals (frame->de.method, TRUE);
-				if (locals) {
+				locals = mono_debug_lookup_locals (frame->de.method);
+				if (locals && CHECK_ICORDBG (FALSE)) {
 					g_assert (pos < locals->num_locals);
 					pos = locals->locals [pos].index;
 					mono_debug_free_locals (locals);
```

Comment:
will this leak locals != null and it is icordbg?

Reply:
Yes. fixed.